cdap-api/src/main/java/co/cask/cdap/api/LocalizationContext.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.api;
+
+import co.cask.cdap.api.annotation.Beta;
+
+import java.io.IOException;
+import java.net.URI;
+import java.util.List;
+
+/**
+ * Context that adds the capability to localize resources in programs.
+ */
+@Beta
+public interface LocalizationContext {
+
+  /**
+   * Localizes a file for the program runtime. Files can be localized in batch programs in their beforeSubmit methods.
+   *
+   * @param uri URI of the file to localize
+   */
+  void addLocalFile(URI uri);
+
+  /**
+   * @return a list of files localized for this program
+   */
+  List<URI> getLocalFiles() throws IOException;
```

Comment:
For this interface to be more useful, I think it should look something like this:

```java
interface LocalizationContext {
  void localize(String name, URI uri);
  void localize(String name, URI uri, boolean archive);

  File getLocalizedFile(String name);
  Map<String, File> getAllLocalizedFiles();
}
```